shell/platform/common/text_input_model.cc

```diff
@@ -79,8 +80,12 @@ void TextInputModel::UpdateComposingText(const std::u16string& text,
       composing_range_.collapsed() ? selection_ : composing_range_;
   text_.replace(rangeToDelete.start(), rangeToDelete.length(), text);
   composing_range_.set_end(composing_range_.start() + text.length());
+#if FML_OS_WIN
```

Comment:
My understanding is that this class is meant to be an OS-agnostic abstraction, so having OS dependant logic here seems weird.

Reply:
This issue (#140739) only occurs on windows.
It seems to have occurred after the modifications below.
https://github.com/flutter/engine/pull/45667

Reply:
Correct - this class is common used by multiple OSes. We should avoid `#ifdef` as much as possible since Flutter is used on many platforms, including some which are not part of our repo -- developers for those embedders can't patch in platform-specific `#ifdef`s so neither should we. (We'd also need platform specific additional testing etc.)

Reply:
I agree that it is not a good idea to include platform branching statements in code in the common area.